### ISSUE TITLE
Support older versions of Kafka

### ISSUE THREAD

**Issue by tsgautier** [reactor/reactor-kafka#18]
What are the chances to add support for Kafka versions:

- 0.10.x
- 0.9.x
- 0.8.x

Not everyone is able to have the latest and greatest Kafka even if they may have the latest and greatest Spring 5 + Reactor + Reactor Kafka.

**Comment by ilayaperumalg** [comment]
Hi @tsgautier,

Thanks for reaching out. 

After discussing with @rajinisivaram on this, here is the reasoning to have the current strategy to support:
```
Kafka broker version 0.10.1.0 and above.
Kafka client version 0.11.0.0 and above.
```

and, this is from @rajinisivaram : 

We upgraded the client to 0.11 since 0.11 clients are compatible with broker version 0.10.1 or later. So we could add transaction support while still supporting older brokers. Broker version of 0.10.1 or higher is required since we rely on consumer protocol changes in that version (and 0.11 clients are not compatible with 0.9 brokers or earlier). To support older versions of broker, reactor-kafka will need to be build with an older version of the clients jar and multiple versions of reactor-kafka will need to be maintained since there were consumer API changes. It is not a worthwhile effort since there have been many bugs in the consumer which were fixed since then.
